ui/src/views/setting/ConfigurationTab.vue

```diff
@@ -0,0 +1,142 @@
+// specific language governing permissions and limitations
+// under the License.
+
+<template>
+  <a-back-top />
+  <a-spin :spinning="tabLoading">
+    <a-table
+      size="small"
+      :showHeader="false"
+      :columns="columns"
+      :dataSource="configdata.filter(config => !config.parent)"
+      :rowKey="record => record.name"
+      :pagination="false"
+      :rowClassName="getRowClassName"
+      style="overflow-y: auto; margin-left: 10px" >
+
+      <template #displaytext="{ record }">
+        <ConfigurationRow :config="configdata" :configrecord="record" :loading="tabLoading" />
+      </template>
+    </a-table>
+  </a-spin>
+</template>
+
+<script>
+import { api } from '@/api'
+import ConfigurationRow from './ConfigurationRow'
+
+export default {
+  name: 'ConfigurationTab',
+  components: {
+    ConfigurationRow
+  },
+  props: {
+    group: {
+      type: String,
+      required: true
+    },
+    subgroup: {
+      type: String,
+      required: false
+    },
+    parent: {
+      type: String,
+      required: false
+    },
+    loading: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      tabLoading: this.loading,
+      columns: [
+        {
+          title: 'Display Text',
+          dataIndex: 'displaytext',
+          slots: { customRender: 'displaytext' }
+        }
+      ],
+      apiName: 'listConfigurations',
+      configdata: []
+    }
+  },
+  created () {
+    this.fetchConfigurationDataByGroup({ group: this.group, subgroup: this.subgroup })
+  },
+  watch: {
+    group: {
+      deep: true,
+      handler (newItem, oldItem) {
+        if (!newItem) {
+          return
+        }
+        this.fetchConfigurationDataByGroup({ group: this.group, subgroup: this.subgroup })
+      }
+    },
+    subgroup: {
+      deep: true,
+      handler (newItem, oldItem) {
+        if (!newItem) {
+          return
+        }
+        this.fetchConfigurationDataByGroup({ group: this.group, subgroup: this.subgroup })
+      }
+    },
+    '$route' (to, from) {
+      if (to.fullPath !== from.fullPath && !to.fullPath.includes('action/')) {
+        if ('name' in to.query) {
+          this.fetchConfigurationDataByGroup({ group: this.group, subgroup: this.subgroup, name: to.query.name })
+        } else {
+          this.fetchConfigurationDataByGroup({ group: this.group, subgroup: this.subgroup })
+        }
+      }
+    },
+    '$i18n.locale' (to, from) {
+      if (to !== from) {
+        this.fetchConfigurationDataByGroup({ group: this.group, subgroup: this.subgroup })
+      }
+    }
+  },
+  methods: {
+    fetchConfigurationDataByGroup (params = {}) {
+      this.tabLoading = true
+      params.pagesize = -1
+      console.log('group name: ' + this.group)
+      api('listConfigurations', params).then(response => {
+        this.configdata = response.listconfigurationsresponse.configuration
+        if (!this.configdata || this.configdata.length === 0) {
```

Comment:
Similar `this.configdata = response?.listconfigurationsresponse?.configuration || []`